Board.cpp

```diff
@@ -217,37 +236,64 @@ set<int> Board::genMoves() const
 			else // pawn moves
 			{
 				int n, m = (side == white) ? 1 : -1;
-				if (squares[i + m * 8].color == none)
+				uint8_t singlePawnMove = static_cast<uint8_t>(i + m * 8);
+				uint8_t doublePawnMove = static_cast<uint8_t>(i + m * 16);
+				bool pawnPromotionWhite = side == white && (i / 8 == 6 );
+					bool pawnPromotionBlack = side == black && (i / 8 == 1 );
+				if (squares[singlePawnMove].color == none)
 				{
-					move.m = {4, i, static_cast<uint8_t>(i + m * 8), 0};
+					move.m = {4, i, static_cast<uint8_t>(singlePawnMove), 0};
```

Comment:
dont need to cast here

Reply:
Yep! Also I like the storing of singlePawnMove and pawnPromotionWhite so you don't have to calculate multiple times.